### DISCUSSION TITLE
`conanDatasourceRegex` doesn't use defined flags

### DISCUSSION THREAD

**Discussion by gost-serb** [renovatebot/renovate#24622] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.106.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Problematic code:
```typescript
export const conanDatasourceRegex = regEx(
  /(?<name>[a-z\-_0-9]+)\/(?<version>[^@/\n]+)(?<userChannel>@\S+\/\S+)/,
  'gim'
);
```
`re2` module doesn't support flags in a separate argument when a pattern is defined as a regular expression literal: either flags must be a part of a pattern definition or pattern must be a regular string.

The only other occurrence of this combination I was able to find in `lib/modules/manager/swift/extract.ts` at line 8:
```typescript
  space: regEx(/(\s+|\/\/[^\n]*|\/\*.*\*\/)+/, 's'),
```

How to reproduce:
```typescript
const RE2 = require('re2');

var r = new RE2(/foo/gim);
var r2 = new RE2(/foo/, 'gim');
var r3 = new RE2("foo", 'gim');

console.log(r.flags)
console.log(r2.flags)
console.log(r3.flags)

```

Proposed solution: move flags to the pattern definition.
If needed I can create PR with the fix.

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by gost-serb** [comment]
Found one more in `lib/util/regex.ts` in `UUIDRegex` definition

**Comment by rarkins** [comment]
PR welcome!

**Reply by gost-serb** [reply, marked as answer]
https://github.com/renovatebot/renovate/pull/24626
